src/ch/ch_monitor.c

```diff
@@ -879,27 +882,33 @@ virCHMonitorResumeVM(virCHMonitorPtr mon)
     return virCHMonitorPutNoContent(mon, URL_VM_RESUME);
 }
 
-int
-virCHMonitorGetCPUInfo(virCHMonitorPtr mon,
-                       virCHMonitorCPUInfoPtr *vcpus,
-                       size_t maxvcpus)
+static void
+virCHMonitorThreadInfoFree(virCHMonitorPtr mon)
+{
+    mon->nthreads = 0;
+    if (mon->threads)
+        VIR_FREE(mon->threads);
```

Comment:
Is this potentially leaking the thrName?

Reply:
@bryteise, can you clarify further how you are expecting a leak here? I don't fully follow.

Above code will clean up threads, if threads is defined. Do you mean in case libvirtd crashes there could be a leak?

Reply:
Ah sorry, re-reading the struct I seemingly ignored the thrName was statically allocated to a fixed size. Sorry about that.